src/e2ir.c

```diff
@@ -1875,6 +1875,68 @@ elem *toElem(Expression *e, IRState *irs)
             return e;
         }
 
+        elem *toElemBinAssign(BinAssignExp *be, int op)
+        {
+            //printf("toElemBinAssign() '%s'\n", be->toChars());
+
+            Type *tb1 = be->e1->type->toBasetype();
+            Type *tb2 = be->e2->type->toBasetype();
+
+            assert(!((tb1->ty == Tarray || tb1->ty == Tsarray ||
+                      tb2->ty == Tarray || tb2->ty == Tsarray) &&
+                     tb2->ty != Tvoid &&
+                     op != OPeq && op != OPandand && op != OPoror));
+
+            tym_t tym = totym(be->type);
+
+            elem *el;
+            elem *ev;
+            if (be->e1->op == TOKcast)
```

Comment:
I don't see why it's necessary, the dmd backend limitation could (should) be worked around in the glue layer so other backends aren't burdened with supporting the invalid ast construct.